pandas/src/interval.pyx

```diff
@@ -101,6 +100,46 @@ cdef class Interval(IntervalMixin):
         end_symbol = ']' if self.closed_right else ')'
         return '%s%s, %s%s' % (start_symbol, self.left, self.right, end_symbol)
 
+    def __add__(self, y):
+        if isinstance(y, numbers.Number):
+            return Interval(self.left + y, self.right + y)
+        elif isinstance(y, Interval) and isinstance(self, numbers.Number):
+            return Interval(y.left + self, y.right + self)
+        else:
+            raise NotImplemented
+
+    def __sub__(self, y):
+        if isinstance(y, numbers.Number):
+            return Interval(self.left - y, self.right - y)
+        elif isinstance(y, Interval) and isinstance(self, numbers.Number):
+            return Interval(y.left - self, y.right - self)
```

Comment:
that said, negating an interval is not going to create another valid interval -- so you can possibly skip this condition.